src/internal/operators/finalize.ts

```diff
@@ -69,7 +69,15 @@ class FinallyOperator<T> implements Operator<T, T> {
   }
 
   call(subscriber: Subscriber<T>, source: any): TeardownLogic {
-    return source.subscribe(new FinallySubscriber(subscriber, this.callback));
+    // The returned subscription will usually be the FinallySubscriber.
+    // However, interop subscribers will be wrapped and for
+    // unsubscriptions to chain correctly, the wrapper needs to be added, too.
+    const finallySubscriber = new FinallySubscriber(subscriber, this.callback);
+    const subscription = source.subscribe(finallySubscriber);
+    if (subscription !== finallySubscriber) {
```

Comment:
I suppose this is to get around edge cases with node where people have more than one version of RxJS?

Reply:
Yep, but the versions can be the same - just different modules. In the related issue, operators from one RxJS module were used with observables from another.

There is another PR that I opened - for discussion - as this is pretty much a general problem: https://github.com/ReactiveX/rxjs/pull/5243

There are a bunch of operators that need `unsubscribe` chained.

Reply:
This is highly debatable. On one hand, I'd like to be ergonomic... On the other hand, it feels like we're just enabling people to build apps with garbage dependency graphs.

On yet ANOTHER hand, Observable is a primitive, like a promise, and people should feel free to export this primitive from their libraries. Ahhh...

Reply:
It's a real problem with Node. The issues opened for this sort of thing are almost always Node-related.

Reply:
Is this something we should just be doing in `Observable` subscribe?

Reply:
I did look at that, but, no. If it's done in `Observable#subscribe`, some tests will fail. I can have a closer look at that if you want an explanation.